### ISSUE TITLE
v23.05.1 fails to compile with `--enable-experimental-features`

### ISSUE THREAD

**Issue by whitslack** [ElementsProject/lightning#6317]
```
wallet/walletrpc.c: In function 'json_transaction_details':
wallet/walletrpc.c:521:56: error: 'const struct wallet_transaction' has no member named 'input_annotations'
  521 |                         struct tx_annotation *ann = &tx->input_annotations[i];
      |                                                        ^~
wallet/walletrpc.c:551:56: error: 'const struct wallet_transaction' has no member named 'output_annotations'
  551 |                         struct tx_annotation *ann = &tx->output_annotations[i];
      |                                                        ^~
```

**Comment by pabpas** [comment]
Same here. These are my options:
```./configure --enable-developer --enable-experimental-features --disable-valgrind```

**Comment by PsySc0rpi0n** [comment]
I think `--enable-experimental-features` is no longer available!
At least the option is no longer available in the help menu of `./configure`.

```
Options include:
  --prefix= (default /usr/local)
    Prefix for make install
  --enable/disable-developer (default disable)
    Developer mode, good for testing
  --enable/disable-compat (default enable)
    Compatibility mode, good to disable to see if your software breaks
  --enable/disable-valgrind (default (autodetect))
    Run tests with Valgrind
  --enable/disable-static (default disable)
    Static link sqlite3 and zlib libraries
  --enable/disable-address-sanitizer (default disable)
    Compile with address-sanitizer
  --enable/disable-ub-sanitizer (default disable)
    Compile with undefined behaviour sanitizer
  --enable/disable-fuzzing (default disable)
    Compile with fuzzing
  --enable/disable-rust (default disable)
    Compile with Rust support
```

**Comment by whitslack** [comment]
> At least the option is no longer available in the help menu of `./configure`.

@PsySc0rpi0n: That's true on `master` but not at `v23.05.1`, which is the release that this issue is addressing.

**Comment by tsjk** [comment]
I stumbled upon the same issue. As far as I remember I had some issues without experimental enabled as some peers needed some feature flag to successfully connect. This means that I cannot upgrade.

**Comment by tsjk** [comment]
As far as I gather annotations were in the process of removal. Now the fields have been too eagerly removed from a header file while they are still used elsewhere. I'd suggest that the CI-pipeline compiles the app with different common combinations of configure flags.

**Comment by tsjk** [comment]
Simply restoring the entries (e.g. https://github.com/tsjk/gentoo-ebuilds/commit/aca47aa378e7db1185a26b5dcd97168f90945488#diff-1d5c540edafd6b86af28c2e57a6a80e9184f47b7dbf10c488f07103bc3d4d84e) makes it compile. No idea if the emitted program is valid though.

**Comment by tsjk** [comment]
Perhaps a better alternative is to include https://github.com/ElementsProject/lightning/commit/b8aa3a579ea03b5eecde44d163fdb12deddf2c8b as well.

**Comment by ksedgwic** [comment]
I'm having this problem too, will try cherry-picking  b8aa3a5 ...

**Comment by ksedgwic** [comment]
Cherry-picking b8aa3a5 works for me.

**Comment by rustyrussell** [comment]
Thanks for this report. This is my mistake, and was not caught by CI since it no longer tests this configuration:(

cherry picking works, though I'll probably go for a subset of that (the walletrpc.c changes), for the next point release

**Comment by rustyrussell** [comment]
BTW I am pretty sure you shouldn't need --experimental-features any more: there are some weird things it enables (quiescence, channel upgrade) but you'd never really notice.  But this should still be fixed.

**Comment by whitslack** [comment]
> I am pretty sure you shouldn't need --experimental-features any more

My Gentoo [ebuild](https://gitlab.com/bitcoin/gentoo/-/blob/47c9694fb9da0adedc59587b8744fcca2a54109e/net-p2p/core-lightning/core-lightning-23.05.1.ebuild#L202) has its `experimental` `USE` flag switched off by default (mirroring the default of your `configure` script), but the flag is still available in 23.05.1 because the option is still available. Maybe no one is enabling the flag, but if they are, they'll run into this compile-time failure. I try to exercise various combinations of `USE` flags (at least a minimal set of flags and all flags but occasionally other combinations as well) to make sure the build succeeds in various configurations.

**Comment by tsjk** [comment]
> > I am pretty sure you shouldn't need --experimental-features any more
> 
> 
> 
> My Gentoo [ebuild](https://gitlab.com/bitcoin/gentoo/-/blob/47c9694fb9da0adedc59587b8744fcca2a54109e/net-p2p/core-lightning/core-lightning-23.05.1.ebuild#L202) has its `experimental` `USE` flag switched off by default (mirroring the default of your `configure` script), but the flag is still available in 23.05.1 because the option is still available. Maybe no one is enabling the flag, but if they are, they'll run into this compile-time failure. I try to exercise various combinations of `USE` flags (at least a minimal set of flags and all flags but occasionally other combinations as well) to make sure the build succeeds in various configurations.

This is exactly how I hit the bug. I use your ebuild! :)

**Comment by rustyrussell** [comment]
Fixed in v23.05.2

Thanks!

**Comment by djmeistro** [comment]
@whitslack hate to try to contact you through this forum so my apologies before hand... I have a channel with bcash_is_trash but my node had a massive force closure with all my channels. My channel to you is inactive but never settled in chain... Any chance you could force close the channel between my node:

035b710d2c8d25993ddcbe031c0fee5434d5e6fb7a6ed5d438404daa99199537b8

Channel point: 778443x1610x4
